### PR TITLE
Add unit tests for Dashboard.

### PR DESCRIPTION
**Pull request by MikeStall**
Test registration,
No functional changes to the product. Some minor changes in startup to aide in test factoring. 

Switch over the DI from AutoFac to Ninject.
- Ninject uses global statics and is harder to test.
- the cool kids recommend AutoFac.

### CODE REVIEW

**Comment by MikeStall** [comment]
I've gone ahead and pushed this commit; but I'll leave this pull request open for a bit longer for comment.

File: src/Dashboard/App_Start/NinjectWebCommon.cs [hunk 1]
```diff
@@ -1,86 +0,0 @@
...
-using System.Web.Http;
-using System.Web.Mvc;
-using Dashboard.AppStart;
-using Dashboard.Infrastructure;
-using Microsoft.Web.Infrastructure.DynamicModuleHelper;
-using Ninject;
-using Ninject.Modules;
-using Ninject.Web.Common;
-
-[assembly: WebActivatorEx.PreApplicationStartMethod(typeof(NinjectWebCommon), "Start")]
```
Comment (mathewc) [inline src/Dashboard/App_Start/NinjectWebCommon.cs]: After removing this, we're no longer using any of the WebActivator/WebActivatorEx packages right? Can we remove those?

File: src/Dashboard/App_Start/NinjectWebCommon.cs [hunk 2]
```diff
@@ -1,86 +0,0 @@
...
-        }
-
-        /// <summary>
-        /// Starts the application
-        /// </summary>
-        [SuppressMessage("Microsoft.Reliability", "CA2000:Dispose objects before losing scope")]
-        public static void Start()
-        {
-            // Registration must occur before application start. So use: WebActivator.PreApplicationStartMethod 
-            DynamicModuleUtility.RegisterModule(typeof(OnePerRequestHttpModule));
```
Comment (mathewc) [inline src/Dashboard/App_Start/NinjectWebCommon.cs]: So none of this startup time logic is needed for AutoFac? It's not clear from looking at the code that all the same behaviors are maintained.

Comment (MikeStall) [inline src/Dashboard/App_Start/NinjectWebCommon.cs]: Correct. That's part of why Autofac won over Ninject.  There's no intrinsic need for any of this.

File: src/Dashboard/App_Start/NinjectWebCommon.cs (hunk 1 above)
Comment (MikeStall) [inline src/Dashboard/App_Start/NinjectWebCommon.cs]: I'll try...

**Comment by mathewc** [comment]
I'm assuming all the Dashboard tests pass with these changes? :)

File: src/Dashboard/App_Start/FilterConfig.cs [hunk 3]
```diff
@@ -15,7 +15,7 @@ public static void RegisterGlobalFilters(GlobalFilterCollection filters)
                 throw new ArgumentNullException("filters");
             }
 
-            filters.Add(new HandleErrorAttribute());
+            //filters.Add(new HandleErrorAttribute());
```
Comment (MikeStall) [inline src/Dashboard/App_Start/FilterConfig.cs]: we can delete the whole file. This is now handled by AutoFac.

File: src/Dashboard/App_Start/NinjectWebCommon.cs (hunk 1 above)
Comment (MikeStall) [inline src/Dashboard/App_Start/NinjectWebCommon.cs]: It looks like we can!

**Comment by MikeStall** [comment]
applied Cr feedback.

**Pull request closed without merging**